en/lessons/basics/basics.md

```diff
@@ -9,269 +9,197 @@ Getting started, basic data types, and basic operations.
+`ghc --version`:
 
-    % elixir -v
-    Erlang/OTP {{ site.erlang.OTP }} [erts-{{ site.erlang.erts }}] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]
-
-    Elixir {{ site.elixir.version }}
+    % ghc --version
+    The Glorious Glasgow Haskell Compilation System, version {{ site.ghc.version }}
 
 ### Trying Interactive Mode
 
-Elixir comes with IEx, an interactive shell, which allows us to evaluate Elixir expressions as we go.
+Your installation should come with `ghci`, an interactive shell that allows you
+to try out code in real time.
 
-To get started, let's run `iex`:
+Follow this tutorial by running `ghci`:
 
-    Erlang/OTP {{ site.erlang.OTP }} [erts-{{ site.erlang.erts }}] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]
+    GHCi, version {{ site.ghc.version }}: https://www.haskell.org/ghc/  :? for help
+    ghci>
 
-    Interactive Elixir ({{ site.elixir.version }}) - press Ctrl+C to exit (type h() ENTER for help)
-    iex>
+To use `Text` in Haskell, type in:
 
-Note: On Windows PowerShell, you need to type `iex.bat`.
+```haskell
+ghci> :set -XOverloadedStrings
+ghci> import qualified Data.Text as T
+```
 
-Let's go ahead and give it a try now by typing in a few simple expressions:
+You can try out Haskell by typing in a few simple expressions:
 
-```elixir
-iex> 2+3
+```haskell
+ghci> 2 + 3
 5
-iex> 2+3 == 5
-true
-iex> String.length("The quick brown fox jumps over the lazy dog")
+ghci> 2 + 3 == 5
+True
+ghci> T.length "The quick brown fox jumps over the lazy dog"
 43
 ```
 
-Don't worry if you don't understand every expression yet, but we hope you get the idea.
+Perhaps you're able to figure out what some of these expressions mean.
 
 ## Basic Data Types
 
 ### Integers
 
-```elixir
-iex> 255
+Haskell has a bounded integer type called `Int`:
+
+```haskell
+ghci> 255 :: Int
 255
 ```
+Note: specify the type to use with `::`.
 
-Support for binary, octal, and hexadecimal numbers comes built in:
+Haskell also supports arbitrary precision integers with `Integer`:
 
-```elixir
-iex> 0b0110
-6
-iex> 0o644
-420
-iex> 0x1F
-31
+```haskell
+ghci> 123456789101112131415
+123456789101112131415
 ```
+Note: `ghci` automatically defaults to `Integer` here.
 
 ### Floats
 
-In Elixir, floating point numbers require a decimal after at least one digit; they have 64-bit double precision and support `e` for exponent values:
+Haskell supports `Double`, a double-precision floating-point number.
+At least one digit must come before the decimal point and Haskell supports
+scientific `e` notation.
 
-```elixir
-iex> 3.14
+```haskell
+ghci> 3.14
 3.14
-iex> .14
-** (SyntaxError) iex:2: syntax error before: '.'
-iex> 1.0e-10
-1.0e-10
-```
+ghci> .14
 
-### Booleans
-
-Elixir supports `true` and `false` as booleans; everything is truthy except for `false` and `nil`:
-
-```elixir
-iex> true
-true
-iex> false
-false
+<interactive>:2:1: error: parse error on input ‘.’
```

Comment:
This doesn't highlight correctly on the generated page. Furthermore, the one line gap inserted (which `ghci` does because a parse error is generated) may be confusing for the reader to parse: line gaps are often used to separate sections of code.

Reply:
Can you show me how it is rendered?

Reply:
This ![image](https://user-images.githubusercontent.com/17554188/120399138-531ba080-c333-11eb-8379-649b33d149ed.png)
rather than this
![image](https://user-images.githubusercontent.com/17554188/120399074-3bdcb300-c333-11eb-99db-6724c0383885.png)